os/board/rtl8730e/src/rtl8730e_mipi_lcdc.c

```diff
@@ -171,11 +173,9 @@ static void rtl8730e_lcd_put_area(u8 *lcd_img_buffer, u32 x_start, u32 y_start,
 #endif
 	LCDC_LayerConfig(pLCDC, LCD_LAYER, &lcdc_init_struct.layerx[LCD_LAYER]);
 	DCache_CleanInvalidate((u32)lcd_img_buffer, LCDC_IMG_BUF_SIZE);
-	lcdc_nextframe = 0;
+	sem_wait(&g_next_frame_block);
+	lcdc_nextframe = 1;
 	LCDC_TrigerSHWReload(pLCDC);
```

Comment:
this flag is checked on ISR, So I thick, This codes is needed to protect critical section using irqsave()

Reply:
In between this codes, If get a IRQ vsync, our sem count can be broken or we can loss the vsync IRQ

Reply:
enter_critical_section() will disable IRQ, this case LCD IRQ will not be trigger, both are waiting for each other.

Reply:
```suggestion
	flag = irqsave()
	lcdc_nextframe = 1;
	LCDC_TrigerSHWReload(pLCDC);
	irqrestore(flag)
```

is it also not commig LCD IRQ??
is the thread waiting in LCDC_TrigerSHWReload() function?

Reply:
I misunderstand your previous point.
This should be working, as only 178 and 179 is protected, should we use enter_critical_section()?